csw-server/src/main/java/org/fao/geonet/kernel/csw/CatalogConfiguration.java

```diff
@@ -180,35 +179,25 @@ private void initDomain(Element operation) {
 	 */
 	private void initDescribeRecordConfig(Element operation) {
 		// Handle typename parameter list value
-		List<Element> typenameList = getTypenamesConfig(operation);
-		
-		String name, prefix, uri, schema;
-		Namespace namespace;
-
-        for (Element typename : typenameList) {
-            name = typename.getAttributeValue(Csw.ConfigFile.Typename.Attr.NAME);
-            prefix = typename.getAttributeValue(Csw.ConfigFile.Typename.Attr.PREFIX);
-            schema = typename.getAttributeValue(Csw.ConfigFile.Typename.Attr.SCHEMA);
-            uri = typename.getAttributeValue(Csw.ConfigFile.Typename.Attr.NAMESPACE);
-            namespace = Namespace.getNamespace(prefix, uri);
-            _describeRecordNamespaces.add(namespace);
-            _describeRecordTypenames.put(prefix + ":" + name, schema);
+        Map<String, Namespace> typenames = _schemaManager.getHmSchemasTypenames();
+        Iterator<String> iterator = typenames.keySet().iterator();
+        while(iterator.hasNext()) {
+            String typeName = iterator.next();
+            Namespace ns = typenames.get(typeName);
+            String typename = typeName;
+            // TODO: Schema plugin schema should be published in
+            // /web/geonetwork/xml/validation/csw/2.0.2 for validation.
```

Comment:
What are the implications about this TODO. I understand the schema xsd should be copied manually to that folder? Can be this improved to do it automatically?